integration/integration_test.go

```diff
@@ -119,7 +121,7 @@ func (s *IntSuite) SetUpSuite(c *check.C) {
 
 	// close & re-open stdin because 'go test' runs with os.stdin connected to /dev/null
 	stdin, err := os.Open("/dev/tty")
-	if err != nil {
+	if err == nil {
```

Comment:
I'm curious if this should just assert `c.Assert(err, check.IsNil)`, or is it an expected scenario, do you know? Kinda weird that tests pass with both this condition and (presumably) erroneous previous condition.

Reply:
This does look suspicious. The original implementation did just that and was changed [here](https://github.com/gravitational/teleport/commit/e4023519d0bea20e187e1666f933fe9ff72e8b7a).

Now, aside from that. the real question is why stdin is required in an integration test? As long as we can input the data from some file, it should be a good replacement for the code that'd require user input otherwise, so it's sort of confusing. I'll look at this in a later PR.

Regarding the logrus update - no, `WithError` will not wrap errors as you imagine since this is gravitational/trace functionality (namely, the TextFormatter implemented therein) which is only available in `trace` 1.1.10 and on. I'll bump it separately.